### PR TITLE
Builder enhancements

### PR DESCRIPTION
**Pull request by razvan**
## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -73,6 +74,198 @@ impl ConfigMapBuilder {
...
+    }
+
+    pub fn run_as_user(&mut self, value: i64) -> &mut Self {
+        self.security_context.run_as_user = Some(value);
+        self
+    }
+
+    pub fn se_linux_level(&mut self, level: &str) -> &mut Self {
+        self.security_context.se_linux_options =
+            Some(self.security_context.se_linux_options.clone().map_or(
```
Comment (teozkr) [inline src/builder.rs:138]: You can use [`.get_or_insert_default(Default::default)`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.get_or_insert_with) to get a `&mut T` from a `&mut Option<T>` without having to write down the logic twice or do any cloning.

File: src/builder.rs [hunk 2]
```diff
@@ -118,6 +260,26 @@ impl ContainerBuilder {
         self
     }
 
+    /// Used for pushing down attributes like the Pod's namespace into the containers.
+    pub fn add_env_var_from_field_ref(
+        &mut self,
+        name: impl Into<String>,
+        value: impl Into<String>,
```
[the remark this answers is not in the document]
Comment (razvan) [reply src/builder.rs]: Done.

File: src/builder.rs [hunk 3]
```diff
@@ -73,6 +75,168 @@ impl ConfigMapBuilder {
...
+}
+
+impl fmt::Display for FieldPathEnvVar {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            FieldPathEnvVar::Name => write!(f, "metadata.name"),
+            FieldPathEnvVar::Namespace => write!(f, "metadata.namespace"),
+            FieldPathEnvVar::UID => write!(f, "metadata.uid"),
+            FieldPathEnvVar::Labels(name) => write!(f, "metadata.labels['{name}']"),
+            FieldPathEnvVar::Annotations(name) => write!(f, "metadata.annotations['{name}']"),
```
Comment (teozkr) [inline src/builder.rs:236]:
```suggestion
            FieldPathEnvVar::Labels(name) => write!(f, "metadata.labels['{}']", name),
            FieldPathEnvVar::Annotations(name) => write!(f, "metadata.annotations['{}']", name),
```

`format!("{foo}")` requires rustc 1.58

**Review by teozkr — APPROVED** [review]

**Comment by razvan** [comment]
bors merge

**Comment by maltesander** [comment]
bors merge

**Comment by razvan** [comment]
bors cancel

**Comment by razvan** [comment]
bors merge

**Pull request merged**